### PR TITLE
Release 2.0.1

### PR DESCRIPTION
**Pull request by sethmlarson**
* [x]  See if all tests, including integration, pass
* [ ]  Get the release pull request approved by a [CODEOWNER](https://github.com/urllib3/urllib3/blob/main/.github/CODEOWNERS)
* [ ]  Squash merge the release pull request with message "`Release <VERSION>`"
* [ ]  Tag with X.Y.Z, push tag on urllib3/urllib3 (not on your fork, update `<REMOTE>` accordingly)
  * Notice that the `<VERSION>` shouldn't have a `v` prefix (Use `1.26.6` instead of `v.1.26.6`)
  * ```
    git tag -s -a '<VERSION>' -m 'Release: <VERSION>'
    git push <REMOTE> --tags
    ```
* [ ]  Execute the `publish` GitHub workflow. This requires a review from a maintainer.
* [ ]  Ensure that all expected artifacts are added to the new GitHub release. Should
       be one `.whl`, one `.tar.gz`, and one `multiple.intoto.jsonl`. Update the GitHub
       release to have the content of the release's changelog.
* [ ]  Announce on:
  * [ ]  Twitter
  * [ ]  Discord
  * [ ]  OpenCollective
* [ ]  Update Tidelift metadata
* [ ]  If this was a 1.26.x release, add changelog to the `main` branch

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -1,3 +1,13 @@
+2.0.0 (2023-04-30)
```
Comment (pquentin) [inline CHANGES.rst:1]:
```suggestion
2.0.1 (2023-04-30)
```

File: CHANGES.rst [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+==================
+
+Fixes
+-----
+
+- Fixed a socket leak when fingerprint or hostname verifications fail. (`#2991 <https://github.com/urllib3/urllib3/issues/2991>`__)
+- Fixed error when ``HTTPResponse.read(0)`` was the first call to read the response body. (`#2998 <https://github.com/urllib3/urllib3/issues/2998>`__)
```
Comment (pquentin) [inline CHANGES.rst:8]: Technically this is not limited to the first call, but I understand how explaining it this way makes it clearer.

**Review by pquentin — CHANGES_REQUESTED** [review]

**Comment by sethmlarson** [comment]
Thanks for the review @pquentin! I've applied the suggested changes.

**Review by pquentin — APPROVED** [review]
Thanks! The previous changelog entry was good, I wasn't requesting a change, to be clear. But this one is good too.

**Pull request merged**